### PR TITLE
Streams on hats tree creation

### PR DESCRIPTION
**Pull request by adamgall**

### CODE REVIEW

File: contracts/DecentHats_0_1_0.sol [hunk 1]
```diff
@@ -113,7 +98,47 @@ contract DecentHats_0_1_0 {
...
+        topHatAccount = createAccount(
+            _registry,
+            _hatsAccountImplementation,
+            salt,
+            address(_hatsProtocol),
+            topHatId
+        );
+    }
+
+    function createHatAndAccountAndMintAndStreams(
```
Comment (mudrila) [inline contracts/DecentHats_0_1_0.sol:141]: Can we rename it to `doTheWholeThingBuddy`?

Comment (mudrila) [reply contracts/DecentHats_0_1_0.sol:141]: Joke ofc 🤣

File: contracts/mocks/MockSablierV2LockupLinear.sol [hunk 2]
```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.19;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../interfaces/sablier/ISablierV2LockupLinear.sol";
+import {LockupLinear} from "../interfaces/sablier/LockupLinear.sol";
+
+contract MockSablierV2LockupLinear is ISablierV2LockupLinear {
```
Comment (mudrila) [inline contracts/mocks/MockSablierV2LockupLinear.sol:8]: Not quite `Mock` lol - this is the whole implementation 🤣

**Review by mudrila — APPROVED** [review]
:shipit:

File: contracts/DecentHats_0_1_0.sol (hunk 1 above)
Comment (adamgall) [reply contracts/DecentHats_0_1_0.sol:141]: lol yeah i know, this function name is kind of insane

File: contracts/mocks/MockSablierV2LockupLinear.sol (hunk 2 above)
Comment (adamgall) [reply contracts/mocks/MockSablierV2LockupLinear.sol:8]: the AI went a little overboard when I asked it to create a mock contract lmao

File: contracts/DecentHats_0_1_0.sol (hunk 1 above)
Comment (DarksightKellar) [reply contracts/DecentHats_0_1_0.sol:141]: slight rename suggestion: `createHatAndAccountAndMintWithStreams`?

**Review by DarksightKellar — APPROVED** [review]

**Pull request merged**